packages/wallet/Account.js

```diff
@@ -0,0 +1,550 @@
+  /**
+   * This is for stellar only
+   * @param {string} assetName
+   * @param {string} issuerPublickey
+   * @param {string} limit
+   * @returns {object|null} xdr.ChangeTrustOp if succeeds or null otherwise
+   */
+  async changeTrust(assetName, issuerPublickey, limit) {
+    switch (this.network) {
+      case 'stellar': {
+        const newAsset = new StellarSdk.Asset(assetName, issuerPublickey);
+        const receiver = await server.loadAccount(this.getAddress());
+        const transaction = new StellarSdk.TransactionBuilder(receiver)
+          .addOperation(StellarSdk.Operation.changeTrust({
+            asset: newAsset,
+            limit
+          }))
+          .build();
+        transaction.sign(this.nativeAccount);
+        return server.submitTransaction(transaction);
+      }
+      case 'ethereum':
+        console.log('Change trust is not for ethereum.');
+        break;
+      default:
+        console.log(setting.networkError);
+    }
+    return null;
+  }
+
+  /**
+  * set the balance field of this account
+  * @param {string} publicKey - the publick key of the account
+  */
+  async loadBalance(publicKey) {
+    try {
+      const response = await rp(this.testAddress + publicKey);
+      this.balance = JSON.parse(response).balances.slice(-1)[0].balance;// XLM balance
+    } catch (err) {
+      console.log('Error in loading balance');
+    }
+  }
+
+  /**
+   * Set the hardware of this account to be the hardware wallet object
+   * If using a hardware, this method must be called right after initialization.
+   * @param {string} hardware - Ledger of Trezor
+   */
+  setHardware(hardware) {
+    switch (hardware) {
+      case 'trezor':
+        this.hardware = new Trezor(this.network);
+        break;
+      case 'ledger':
+        this.hardware = new Ledger(this.network);
+        break;
+      default:
+        console.log('The hardware must be ledger or trezor');
+        return false;
+    }
+    return true;
+  }
+
+  /**
+   * Remove the hardware, and set the account using native account objects
+   */
+  removeHardware() {
+    this.hardware = null;
+  }
+
+  /**
+   * Set the account field of this account to be the argument;
+   * For Stellar, upload it to the testnet and update the balance;
+   * For Ethereum, update its balance read on the Rinbeky testnet.
+   * @param {object|string} account - A stellar or ethereum account
+   * or the derivation path of the account in the hardware wallet
+   * If using a hardware wallet, the nativeAccount field will be the derivation path
+   * This is for the convenience of using Ledger/Trezor APIs.
+   * The public key will be available in getAccount method,
+   * and the private key of the account is not released by the wallet.
+   */
+  async setAccount(account) {
+    if (this.hardware != null) {
+      switch (this.hardware.hardware) {
+        case 'ledger':
+        case 'trezor':
+          // they are the same initialization
+          this.nativeAccount = account;
```

Comment:
`setHardware` and `removeHardware` should be called in `setAccount` depending on the account being set and whether existing hardware wallet is present